scenarios/bring-your-own-data/bringYourOwnData.md

```diff
@@ -0,0 +1,126 @@
+# Use Azure OpenAI On Your Data with openai
+
+## Objective
+
+The main objective of this tutorial is to help users understand the process of chatting on top of your own data. By the end of this tutorial, you should be able to:
+
+ - Connect to Azure using AzureOpenAI client
+ - Get chat response with your own data
+
+## Time
+
+You should expect to spend 5-10 minutes running this sample. 
+
+## Before you begin
+
+### Set environment variables
+
+To start with let us create a config `.env` file with your project details. This file can be used in this sample or other samples to connect to your workspace. A sample `.env` file is provided below with the variables that you need.
+
+```js
+OPENAI_API_VERSION="Insert the desired Azure OpenAI API version here"
+AZURE_OPENAI_ENDPOINT="Insert your Azure OpenAI resource endpoint here"
+AZURE_OPENAI_API_KEY="Insert your API Key here"
+
+AZURE_SEARCH_ENDPOINT="Insert your Search Endpoint here"
+AZURE_SEARCH_INDEX="Insert your Search index name here"
+AZURE_SEARCH_KEY="Insert your Search API Key here"
+```
+
+### Connect to your data source
+We need to connect to a data source to upload our data. We will set up your own designated data sources using Azure AI Search and import the values from our `.env` file.
+
+```js
+require('dotenv').config();
+
+const azureSearchEndpoint = process.env["AZURE_SEARCH_ENDPOINT"] || "<search endpoint>";
+const azureSearchIndexName = process.env["AZURE_SEARCH_INDEX"] || "<search index>";
+const azureSearchAdminKey = process.env["AZURE_SEARCH_KEY"] || "<search key>";
+
+const dataSources = {
+    data_sources: [
+        {
+            type: "azure_search",
+            parameters: {
+                endpoint: azureSearchEndpoint,
+                index_name: azureSearchIndexName,
+                authentication: {
+                    type: "api_key",
```

Comment:
Let's replace API key with managed identity. We don't want to encourage customers to store a secret (the key) somewhere when managed identity is available.